### ISSUE TITLE
U-2-Net for binary segmentation

### ISSUE THREAD

**Issue by dennisbappert** [NathanUA/U-2-Net#18]
Hey @NathanUA,

just to share my experience here, using the small U-2-Net architecture:
- I'm comparing the results to a baseline model (DeepLabV3, ResNet101 ~450MB) and achieved 82,... mIOU after 500 Epochs on my rather small benchmarking Dataset (license plate binary segmentation).
- The small U-2-Net model achieved 78,... mIOU (~5MB) after roughly 600 Epochs

I did the following modifications to the model in order to fine-tune my results:
- ~I introduced a distillation loss between each nested Unet and used temperature annealing on the sigmoid. Based on the assumption that the more nested Unets have more computational power, we can define a second loss specifying the bce between adjacent layers. The model converges way faster with this approach, however the 3 new hyperparameters ("temperature", alpha and a scaling factor for the loss) seem to be quite "sensitive".~ was not working.
- I changed to SGD with a poly learning rate starting at 0.01
- I'm currently exploring the options to further prune the network or use an early exit while inferencing to reduce inferencing time.

I just wanted to share my experience here, maybe it can be helpful for this repository.

Best,
Dennis

**Comment by NathanUA** [comment]
Hi, Dennis, 

Thanks a lot for your trying. Actually, we don't think that our currently used setting is the best option. The performance of U^2-Net can be further improved by tuning these parameters. I think your experiences are very helpful for others who want to use U^2-Net for their own projects. **You can also try to change the filter numbers of the U^2-Net to instantiate your own U^2-Net version. Besides, the elementary conv-bn-relu function can also be replace by residual, inception, dilated convolution and even another small unet (or rsu) block, etc.** 

Regards,
Xuebin

**Comment by yxt132** [comment]
do you think SGD may perform better than ADAM?

**Comment by johnyquest7** [comment]
@dennisbappert could you please share your code using SGD? Interested in seeing your experiments
